types/react-router/index.d.ts

```diff
@@ -146,6 +146,56 @@ export function matchPath<Params extends { [K in keyof Params]?: string }>(
     parent?: match<Params> | null,
 ): match<Params> | null;
 
+// Must only use abcdefghijklmnopqrstuvwxyz0123456789-_|
+export type IsExtractableRegex<T extends string, L = Lowercase<T>> =
```

Comment:
Since we can't use TS version 4.5, I can't think of a cleaner way to check if it's alphanumeric.

Previously I did 
```
type ExtractCharacters<T extends string> = T extends `${infer Left}${infer Right}`
    ? Left | ExtractCharacters<Right>
    : T;

type OnlyIncludes<T extends string, Allowed extends string> = ExtractCharacters<T> extends ExtractCharacters<Allowed>
    ? true
    : false;
```

But `ExtractCharacters` gives a "too deep" error in version 4.4 and below

```
ERROR: 155:45  expect                           Compile error in typescript@4.4 but not in typescript@4.5.
Fix with a comment '// Minimum TypeScript Version: 4.5' just under the header.
Type instantiation is excessively deep and possibly infinite.
```

Understandable if you don't want to maintain this.

Reply:
We should get the opinion of some TypeScript maintainers with regards to perf. It's not obvious to me that a 26x recursive `T extends A ? true : recurse()` doesn't have a significant perf impact.

Reply:
Good point. I did some testing in TS playground with longer paths (100+ characters) and it registers the typing in under a second, feels the same as without the change. Not a surefire test, but hopefully that helps.

Reply:
I really cannot endorse this approach. This is going to result in a ton of conditional type evaluations and is likely to be quite expensive in heavy use.

Reply:
No worries, thanks for reviewing 👍